File: Auth/Java/app/src/main/java/com/fj/tpamobile/auth_java/RegisterActivity.java
package com.fj.tpamobile.auth_java;

import android.support.annotation.NonNull;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.OnFailureListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.auth.AuthResult;
import com.google.firebase.auth.FirebaseAuth;

public class RegisterActivity extends AppCompatActivity implements View.OnClickListener {

    private EditText edtEmail, edtPassword;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_register);

        Button btnRegister = findViewById(R.id.btn_register);
        edtEmail = findViewById(R.id.edt_email);
        edtPassword = findViewById(R.id.edt_password);

        btnRegister.setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()){
            case R.id.btn_register:
                String email = edtEmail.getText().toString();
                String password = edtPassword.getText().toString();

                if (email.isEmpty()){
                    edtEmail.setError("Must be filled");
                }else if (password.isEmpty()){
                    edtPassword.setError("Must be filled");
                }else{
                    FirebaseAuth auth = FirebaseAuth.getInstance();

                    auth.createUserWithEmailAndPassword(email, password)
                            .addOnCompleteListener(new OnCompleteListener<AuthResult>() {
                                @Override
                                public void onComplete(@NonNull Task<AuthResult> task) {
                                    if (task.isSuccessful()){
                                        Toast.makeText(RegisterActivity.this, "Successfully", Toast.LENGTH_SHORT).show();

                                        finish();
                                    }else{
                                        Toast.makeText(RegisterActivity.this, "Email already registered", Toast.LENGTH_SHORT).show();
                                    }
                                }
                            })
                            .addOnFailureListener(new OnFailureListener() {
                                @Override
                                public void onFailure(@NonNull Exception e) {
                                    Log.d("InfoRegister", e.getMessage());
                                }
                            });
                }
                break;
        }
    }
}

File: Auth/Kotlin/app/src/main/java/com/fj/tpamobile/auth_kotlin/RegisterActivity.kt
package com.fj.tpamobile.auth_kotlin

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import com.google.firebase.auth.FirebaseAuth
import kotlinx.android.synthetic.main.activity_register.*
import org.jetbrains.anko.toast

class RegisterActivity : AppCompatActivity(), View.OnClickListener {
    override fun onClick(v: View?) {
        when(v){
            btn_register -> {
                val email = edt_email.text.toString()
                val password = edt_password.text.toString()

                when {
                    email.isEmpty() -> edt_email.error = "Must be filled"
                    password.isEmpty() -> edt_password.error = "Must be filled"
                    else -> {
                        val auth = FirebaseAuth.getInstance()

                        auth.createUserWithEmailAndPassword(email, password)
                            .addOnCompleteListener(this){
                                if (it.isSuccessful){
                                    toast("Successfully")

                                    finish()
                                }else{
                                    toast("Email already registered!")
                                }
                            }
                            .addOnFailureListener(this){
                                println("InfoRegister: ${it.message}")
                            }
                    }
                }
            }
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_register)

        btn_register.setOnClickListener(this)
    }
}
